process/transactionProcessor.go

```diff
@@ -87,13 +90,22 @@ func NewTransactionProcessor(
 	if newTxCostProcessor == nil {
 		return nil, ErrNilNewTxCostHandlerFunc
 	}
+	jsonMarshalizer, err := marshalFactory.NewMarshalizer("json")
+	if err != nil {
+		return nil, err
+	}
+	logsMerger, err := logsevents.NewLogsMerger(hasher, jsonMarshalizer)
+	if err != nil {
+		return nil, err
+	}
 
 	return &TransactionProcessor{
 		proc:               proc,
 		pubKeyConverter:    pubKeyConverter,
 		hasher:             hasher,
 		marshalizer:        marshalizer,
 		newTxCostProcessor: newTxCostProcessor,
+		mergeLogsHandler:   logsMerger,
```

Comment:
refactor this struct:
-> use args for passing data into the constructor
-> send logsMerger as an argument instead of initializing it on the constructor

Reply:
done. now we have a factory for the transaction processor